backend/src/grouter/rdp.c

```diff
@@ -326,10 +329,17 @@ void rdp_gobackn_recv_callback (
     if(is_ack_packet) {
     	//If it is an ack packet and for the sequence number we are waiting for then we update the context
     	if(seq_num == seq_num_expected){
-
+    		if(gbn_context->waiting) {
+    			printf("Got acknowledgement.\n");
+    			
+    		}
+    		gbn_context->pcb_start = (gbn_context->pcb_start + 1) % MAX_N_CALLBACK;
```

Comment:
If i remember correctly, gobackn does cumulative acks so I think even if we get an ack for something bigger than the base but still in our window it counts as an ack for everything since the base. So this should be sequence number received + 1 mod N I think